include/Board.h

```diff
@@ -1,20 +1,43 @@
-#pragma once
-#include "Piece.h"
-#include <array>
-#include "memory"
-
-using std::array;
-using std::unique_ptr;
-const auto START_BOARD = "RNBQKBNRPPPPPPPP################################pppppppprnbqkbnr";
-const auto BOARD_SIZE = 8;
-
-enum class Type { King, Queen, Rook, Bishop, Knight, Pawn, EmptySlot, Unknown };
-
-class Board {
-public:
-	Board(const string& start = START_BOARD);
-	string getBoard() const;
-
-private:
-	array <array< unique_ptr<Piece>, BOARD_SIZE> ,BOARD_SIZE> m_board;
+#pragma once
+#include "Piece.h"
+#include <array>
+#include "memory"
+#include <string>
+using std::string;
+using std::array;
+using std::unique_ptr;
+
+const auto START_BOARD = string("RNBQKBNRPPPPPPPP################################pppppppprnbqkbnr");
+const auto BOARD_SIZE = 8;
+// ============================== CODE RESPONCE ENUM ==================
+enum CodeRes : int {
+	EmptyPiece = 11,
+	NoteYourPiece = 12,
+	YourPieceFoundAtDest = 13,
+	IlligalMove = 21,
+	CheckMateOnMyself = 31,
+	CheckMateOnMOpponent = 41,
+	LegalMove = 42,
+};
+
+//=====================================================================
+class Board {
+public:
+	// =========================== C-TOR ==============================
+	Board(const string& start = START_BOARD);
+
+	// =========================== METHODS ============================
+
+	int getCodeResponse(const string& inpt, PlayerColor playerTurn);
+    const auto& getStringBoard() const{ return m_stringBoard;}

```

Comment:
this method returns a const reference to the m_stringBoard member, but it should return a copy or a const reference to a const member. This can prevent the caller from modifying the board unintentionally.